### PR TITLE
feat: Add `NullableFromJSONSchemaTags` option

### PR DESCRIPTION
**Pull request by candiduslynx**
This changes default behavior to look for the `jsonschema:"nullable"` tag to mark field optional only `Reflector.NullableFromJSONSchemaTags` is set to `true`.

In Go it's perfectly fine to unmarshal the following types from `null`:
- pointers
- slices
- maps
- interfaces

This PR also includes the changes from #107, so it might be better to merge that one first.

### CODE REVIEW

**Comment by candiduslynx** [comment]
@samlown this changes the default behavior to be closer to default `encoding/json` implementation.

The other option could be make the default behavior to look for tags & only explicitly use field type for nullability (when some option is set).

File: fixtures/allow_additional_props.json [hunk 1]
```diff
@@ -162,9 +162,9 @@
         "mult": {
           "type": "number",
           "enum": [
-            1,
+            1.0,
```
Comment (candiduslynx) [inline fixtures/allow_additional_props.json:165]: these are just result of running `go test -update ./...`

File: fixtures/defaults_expanded_toplevel.json [hunk 2]
```diff
@@ -162,9 +162,9 @@
     "mult": {
       "type": "number",
       "enum": [
-        1,
+        1.0,
```
Comment (candiduslynx) [inline fixtures/defaults_expanded_toplevel.json:165]: these are just result of running `go test -update ./...`

File: fixtures/ignore_type.json [hunk 3]
```diff
@@ -156,9 +156,9 @@
         "mult": {
           "type": "number",
           "enum": [
-            1,
+            1.0,
```
Comment (candiduslynx) [inline fixtures/ignore_type.json:159]: these are just result of running `go test -update ./...`

File: fixtures/no_reference.json [hunk 4]
```diff
@@ -151,9 +151,9 @@
     "mult": {
       "type": "number",
       "enum": [
-        1,
+        1.0,
```
Comment (candiduslynx) [inline fixtures/no_reference.json:154]: these are just result of running `go test -update ./...`

**Comment by samlown** [comment]
@candiduslynx thanks for this, but I'm trying to understand what exact problem you're trying to solve with this PR. For me, the nullable type is implied when there is no value, I'm really not sure why you'd want to define it explicitly.

File: fixtures/allow_additional_props.json (hunk 1 above)
Comment (samlown) [reply fixtures/allow_additional_props.json:165]: I'm not happy about those... these slipped the net. Will try and figure out what happened there.

Comment (samlown) [reply fixtures/allow_additional_props.json:165]: Okay, the new output is correct and aligned with expectations. The main branch is now up to date which should simplify this PR.

**Comment by candiduslynx** [comment]
@samlown consider json `{"a":null} `.
If schema defines property `a` having `"type": "object" `, the validation will fail.
But the `{} ` might validate successfully, if `a` isn't a required property.
So, for structs that don't have `json:",omitempty"` a perfectly valid marshaled representation with explicit `null` value will fail.

**Comment by candiduslynx** [comment]
@samlown This PR also includes the changes from https://github.com/invopop/jsonschema/pull/107, so it might be better to merge that one first.

**Comment by candiduslynx** [comment]
I'm going to reopen a fresh version that preserves default behavior + add README.md doc for this

**Comment by candiduslynx** [comment]
Replaced by #110

**Pull request closed without merging**